### PR TITLE
storage service endpoint

### PR DESCRIPTION
**Pull request by OrGur1987**
see issue https://github.com/ManageIQ/manageiq-ui-classic/issues/8616

### CODE REVIEW

**Comment by OrGur1987** [comment]
https://codeclimate.com/github/ManageIQ/manageiq-api/pull/1198
this is just 2 lines of dup code, can we ignore it?

**Comment by OrGur1987** [comment]
hi @kbrock, 
maybe this requires a cross-repo check with the main repo for `db/fixtures/miq_product_features.yml`

**Comment by OrGur1987** [comment]
@miq-bot cross-repo-tests https://github.com/ManageIQ/manageiq/pull/22331

**Comment by agrare** [comment]
@OrGur1987 there are some storage_service related failures in the cross-repo tests.  Are the API specs green for you locally with that core PR applied?

**Comment by OrGur1987** [comment]
> @OrGur1987 there are some storage_service related failures in the cross-repo tests. Are the API specs green for you locally with that core PR applied?

yes, perhaps I need to list all the PRs from this issue and not only https://github.com/ManageIQ/manageiq/pull/22331 ?

**Comment by agrare** [comment]
I'm guessing it is because your autoade provider PR wasn't included and that is causing supports failures.  That indicates a different kind of issue but I'll comment in the diff

File: spec/requests/storage_services_spec.rb [hunk 1]
```diff
@@ -30,4 +30,49 @@
...
+        "action"   => "create",
+        "resource" => {
+          "ems_id"      => provider.id,
+          "name"        => "test_storage_service",
+          "description" => "description of test_storage_service",
+        }
+      }
+      post(api_storage_services_url, :params => request)
+      expect_multiple_action_result(1, :success => true, :message => /Creating Storage Service test_storage_service for Provider #{provider.name}/, :task => true)
+    end
```
Comment (agrare) [inline spec/requests/storage_services_spec.rb:48]: So the issue we have here is we are depending on the actual class supports values which limits us quite a bit.

The preferred way to do this is to use `stub_supports` to control supports/supports_not without having to change types.

Add the SupportsHelper to this spec:
```
 describe "Storage Services API" do
+  include Spec::Support::SupportsHelper
+
```

Then you can `stub_supports(StorageService, :create)` and `stub_supports(service, :delete)`

File: spec/requests/storage_services_spec.rb [hunk 2]
```diff
@@ -30,4 +30,49 @@
       expect(response.parsed_body).to include(expected)
     end
   end
+
+  context "POST /api/host_initiators" do
+    it "creates new Storage Service" do
+      api_basic_authorize(collection_action_identifier(:storage_services, :create))
+      provider = FactoryBot.create(:ems_autosde, :name => 'Autosde')
```
Comment (agrare) [inline spec/requests/storage_services_spec.rb:37]: Unless the name is required best to leave it off
```suggestion
      provider = FactoryBot.create(:ems_autosde)
```

File: spec/requests/storage_services_spec.rb (hunk 1 above)
Comment (OrGur1987) [reply spec/requests/storage_services_spec.rb:48]: should I just add the calls to `stub_supports` before the post requests for create/delete? say in lines 46 and 56...

**Comment by OrGur1987** [comment]
@miq-bot cross-repo-tests https://github.com/ManageIQ/manageiq/pull/22331

File: spec/requests/storage_services_spec.rb (hunk 1 above)
Comment (agrare) [reply spec/requests/storage_services_spec.rb]: Yeah looks good

**Comment by agrare** [comment]
Just the normal authentication_spec failures in cross-repo, not related to this PR.  Merged the Core PR and kicked specs

**Review by agrare — APPROVED** [review]

**Pull request merged**